optuna/study/_optimize.py

```diff
@@ -237,93 +222,38 @@ def _run_trial(
 
     # `Study.tell` may raise during trial post-processing.
     try:
-        study.tell(trial, values=values, state=state)
+        with warnings.catch_warnings(record=True) as tell_warnings:
+            warnings.simplefilter("always", UserWarning)
```

Comment:
```
Once the context manager exits, the warnings filter is restored to its state when the context was entered.
```
https://docs.python.org/3/library/warnings.html#testing-warnings

Reply:
Should we try not to rely on this logic given https://bugs.python.org/issue29672 ?

Reply:
I missed your comment. :bow: Yes, I think so.